plugins/inputs/smart/README.md

```diff
@@ -0,0 +1,103 @@
+# Telegraf S.M.A.R.T. plugin
+
+Get metrics using the command line utility `smartctl` for S.M.A.R.T. (Self-Monitoring, Analysis and Reporting Technology) storage devices. SMART is a monitoring system included in computer hard disk drives (HDDs) and solid-state drives (SSDs)[1] that detects and reports on various indicators of drive reliability, with the intent of enabling the anticipation of hardware failures.
+See smartmontools (https://www.smartmontools.org/).
+
+If no devices are specified, the plugin will scan for SMART devices via the following command:
+
+```
+smartctl --scan
+```
+
+Metrics will be reported from the following `smartctl` command:
+
+```
+smartctl --info --attributes --health -n <nocheck> --format=brief <device>
```

Comment:
This is actually not generating any output for the attribute checker on RHEL7 using smartctl 6.2. See below...

```
[telegraf@carf-metrics-influx02 ~]$ ./tester --info --attributes --health --format=brief /dev/sdcp
smartctl 6.2 2013-07-26 r3841 [x86_64-linux-3.10.0-514.6.1.el7.jump1.x86_64] (local build)
Copyright (C) 2002-13, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Vendor:               TOSHIBA
Product:              PX02SMF040
Revision:             A3B3
User Capacity:        400,088,457,216 bytes [400 GB]
Logical block size:   512 bytes
LU is resource provisioned, LBPRZ=1
Rotation Rate:        Solid State Device
Form Factor:          2.5 inches
Logical Unit id:      0x500003965c89f4a0
Serial number:        65J0A025T0QB
Device type:          disk
Transport protocol:   SAS
Local Time is:        Thu Apr  6 17:36:40 2017 CDT
SMART support is:     Available - device has SMART capability.
SMART support is:     Enabled
Temperature Warning:  Disabled or Not Supported

=== START OF READ SMART DATA SECTION ===
SMART Health Status: OK

SS Media used endurance indicator: 0%
Current Drive Temperature:     38 C
Drive Trip Temperature:        60 C

Manufactured in week 25 of year 2015
Elements in grown defect list: 0

[telegraf@carf-metrics-influx02 ~]$ ./tester --info -x --attributes --health --format=brief /dev/sdcp
smartctl 6.2 2013-07-26 r3841 [x86_64-linux-3.10.0-514.6.1.el7.jump1.x86_64] (local build)
Copyright (C) 2002-13, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Vendor:               TOSHIBA
Product:              PX02SMF040
Revision:             A3B3
User Capacity:        400,088,457,216 bytes [400 GB]
Logical block size:   512 bytes
LU is resource provisioned, LBPRZ=1
Rotation Rate:        Solid State Device
Form Factor:          2.5 inches
Logical Unit id:      0x500003965c89f4a0
Serial number:        65J0A025T0QB
Device type:          disk
Transport protocol:   SAS
Local Time is:        Thu Apr  6 17:37:21 2017 CDT
SMART support is:     Available - device has SMART capability.
SMART support is:     Enabled
Temperature Warning:  Disabled or Not Supported
Read Cache is:        Enabled
Writeback Cache is:   Disabled

=== START OF READ SMART DATA SECTION ===
SMART Health Status: OK

SS Media used endurance indicator: 0%
Current Drive Temperature:     38 C
Drive Trip Temperature:        60 C

Manufactured in week 25 of year 2015
Elements in grown defect list: 0

Error counter log:
           Errors Corrected by           Total   Correction     Gigabytes    Total
               ECC          rereads/    errors   algorithm      processed    uncorrected
           fast | delayed   rewrites  corrected  invocations   [10^9 bytes]  errors
read:          0        0         0         0          0      33119.546           0
write:         0        0         0         0          0      74858.031           0
verify:        0        0         0         0          0          2.622           0

Non-medium error count:       32

SMART Self-test log
Num  Test              Status                 segment  LifeTime  LBA_first_err [SK ASC ASQ]
     Description                              number   (hours)
# 1  Background short  Completed                  64       3                 - [-   -    -]
# 2  Background long   Completed                  64       2                 - [-   -    -]
# 3  Background short  Completed                  64       2                 - [-   -    -]
Long (extended) Self Test duration: 1800 seconds [30.0 minutes]

Device does not support Background scan results logging
Protocol Specific port log page for SAS SSP
relative target port id = 1
  generation code = 4
  number of phys = 1
  phy identifier = 0
    attached device type: expander device
    attached reason: SMP phy control function
    reason: unknown
    negotiated logical link rate: reserved [11]
    attached initiator port: ssp=0 stp=0 smp=0
    attached target port: ssp=0 stp=0 smp=1
    SAS address = 0x500003965c89f4a2
    attached SAS address = 0x5f8db882fbf5737f
    attached phy identifier = 6
    Invalid DWORD count = 4
    Running disparity error count = 3
    Loss of DWORD synchronization = 0
    Phy reset problem = 0
relative target port id = 2
  generation code = 4
  number of phys = 1
  phy identifier = 1
    attached device type: expander device
    attached reason: SMP phy control function
    reason: loss of dword synchronization
    negotiated logical link rate: reserved [11]
    attached initiator port: ssp=0 stp=0 smp=0
    attached target port: ssp=0 stp=0 smp=1
    SAS address = 0x500003965c89f4a3
    attached SAS address = 0x5f8db882fbf573ff
    attached phy identifier = 6
    Invalid DWORD count = 0
    Running disparity error count = 0
    Loss of DWORD synchronization = 0
    Phy reset problem = 0
Only support protocol specific log page on SAS devices

```